examples/model_selection/plot_roc.py

```diff
@@ -1,132 +1,269 @@
+can be defined unambiguously. In the case of multiclass classification, a notion
+of TPR/FPR is obtained only after binarizing the output. This can be done in 2
+different ways:
 
-Another evaluation measure for multi-label classification is
-macro-averaging, which gives equal weight to the classification of each
-label.
+- the One-vs-Rest scheme compares each class against all the others (assumed as
+  one);
+- the One-vs-One scheme compares every unique pairwise combination of classes.
 
-.. note::
+In this example we explore both schemes and demo the concepts of micro and macro
+averaging as different ways of summarizing the information of the multiclass ROC
+curves.
 
-    See also :func:`sklearn.metrics.roc_auc_score`,
-             :ref:`sphx_glr_auto_examples_model_selection_plot_roc_crossval.py`
+.. note::
 
+    See :ref:`sphx_glr_auto_examples_model_selection_plot_roc_crossval.py` for
+    an extension of the present example estimating the variance of the ROC
+    curves and their respective AUC.
 """
 
-import numpy as np
-import matplotlib.pyplot as plt
-from itertools import cycle
+# %%
+# Load and prepare data
+# =====================
+#
+# We import the :ref:`iris_dataset` which contains 3 classes, each one
+# corresponding to a type of iris plant. One class is linearly separable from
+# the other 2; the latter are **not** linearly separable from each other.
+#
+# Here we binarize the output and add noisy features to make the problem harder.
 
-from sklearn import svm, datasets
-from sklearn.metrics import roc_curve, auc
+import numpy as np
+from sklearn.datasets import load_iris
 from sklearn.model_selection import train_test_split
-from sklearn.preprocessing import label_binarize
-from sklearn.multiclass import OneVsRestClassifier
-from sklearn.metrics import roc_auc_score
-
-# Import some data to play with
-iris = datasets.load_iris()
-X = iris.data
-y = iris.target
 
-# Binarize the output
-y = label_binarize(y, classes=[0, 1, 2])
-n_classes = y.shape[1]
+iris = load_iris()
+target_names = iris.target_names
+X, y = iris.data, iris.target
+y = iris.target_names[y]
 
-# Add noisy features to make the problem harder
 random_state = np.random.RandomState(0)
 n_samples, n_features = X.shape
-X = np.c_[X, random_state.randn(n_samples, 200 * n_features)]
+n_classes = len(np.unique(y))
+X = np.concatenate([X, random_state.randn(n_samples, 200 * n_features)], axis=1)
+(
+    X_train,
+    X_test,
+    y_train,
+    y_test,
+) = train_test_split(X, y, test_size=0.5, stratify=y, random_state=0)
 
-# shuffle and split training and test sets
-X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.5, random_state=0)
+# %%
+# We train a :class:`~sklearn.linear_model.LogisticRegression` model which can
+# naturally handle multiclass problems, thanks to the use of the multinomial
+# formulation. It is also possible to use OvR ROC evaluation of an OvO
+# multiclass support vector machine model such as :class:`~sklearn.svm.SVC`.
```

Comment:
Let's not confuse the user at this point.

```suggestion
# formulation.
```

The distinction between OvR vs OvO for evaluation and OvR vs OvO for training is better clarified in the note introduced in the next section.